packages/native-stack/src/views/DebugContainer.native.tsx

```diff
@@ -9,7 +9,12 @@ type ContainerProps = ViewProps & {
   children: React.ReactNode;
 };
 
-let Container = View as unknown as React.ComponentType<ContainerProps>;
+// This view must *not* be flattened.
+// See https://github.com/software-mansion/react-native-screens/pull/1825
+// for detailed explanation.
+let Container = function (props: ContainerProps): JSX.Element {
+  return <View {...props} collapsable={false} />;
+};
```

Comment:
I'm not a typescript chad by any means, so please pay attention to whether the type of the `Container` has changed. It was earlier `React.ComponentType<ContainerProps>`.

The same applies in mirror PR to v7

* https://github.com/react-navigation/react-navigation/pull/11501/